### ISSUE TITLE
Expose envoy healthcheck endpoint on traffic port

### ISSUE THREAD

**Issue by haswalt** [saarasio/enroute#73]
External load balancers providers like AWS require a healthcheck in order for hosts to be enabled in the pool. Typically it is expected that a healthcheck is exposed by the target-port, usually this is a simple 200 or 204 response to `GET /`. However unless there is a specific route configured EnRoute / Envoy returns 404 for these checks resulting in hosts being removed from the pool.

Ideally to check the host status the Envoy healthcheck should be exposed, this can be achieved by adding an additional port mapping to Envoy (port `8000`):

```
spec:
  ports:
  - name: http
    port: 80
    protocol: TCP
    targetPort: 8080
  - name: https
    port: 443
    protocol: TCP
    targetPort: 8080
  - name: hc
    port: 8000
    protocol: TCP
    targetPort: 8000
```

However this exposes other Envoy endpoints to the public, EnRoute should include it's own healtcheck endpoint that either proxies or checks the Envoy status before return a `200`.

**Comment by chintan8saaras** [comment]
Helm chart changes related to this issue is tracked here #72

HealthCheck is active on the configured listener port. Specifying the healthcheck path is configurable in filter config. Below is an example of working HealthCheck filter.


```

$ kubectl describe -n httpbin httpfilters.enroute.saaras.io httpbin-80-healthcheckfilter
Name:         httpbin-80-healthcheckfilter
Namespace:    httpbin
Labels:       app=httpbin-app
              app.kubernetes.io/managed-by=Helm
Annotations:  meta.helm.sh/release-name: httpbin-service-policy
              meta.helm.sh/release-namespace: httpbin
API Version:  enroute.saaras.io/v1
Kind:         HttpFilter
Metadata:
  Creation Timestamp:  2022-07-18T22:54:22Z
  Generation:          1
  Managed Fields:
    API Version:  enroute.saaras.io/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:meta.helm.sh/release-name:
          f:meta.helm.sh/release-namespace:
        f:labels:
          .:
          f:app:
          f:app.kubernetes.io/managed-by:
      f:spec:
        .:
        f:httpFilterConfig:
          .:
          f:config:
        f:name:
        f:type:
    Manager:         helm
    Operation:       Update
    Time:            2022-07-18T22:54:22Z
  Resource Version:  22610
  UID:               a31724c3-a896-43ee-8c74-16beeb51baef
Spec:
  Http Filter Config:
    Config:  {
    "path" : "/healthz"
}

  Name:  httpbin-80-healtcheck
  Type:  http_filter_healthcheck
Events:  <none>
$ curl -v 212.2.244.185/healthz
*   Trying 212.2.244.185:80...
* TCP_NODELAY set
* Connected to 212.2.244.185 (212.2.244.185) port 80 (#0)
> GET /healthz HTTP/1.1
> Host: 212.2.244.185
> User-Agent: curl/7.68.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< vary: Accept-Encoding
< lua-filter-says: Hello
< x-envoy-upstream-healthchecked-cluster: enroutedemo
< date: Tue, 19 Jul 2022 04:18:41 GMT
< server: envoy
< content-length: 0
<
* Connection #0 to host 212.2.244.185 left intact 
```

**Comment by chintan8saaras** [comment]
This got resolved in release [v0.11.0](https://github.com/saarasio/enroute/releases/tag/v0.11.0)
